src/zenml/service_connectors/service_connector.py

```diff
@@ -1128,98 +1129,149 @@ def verify(
+                    require_resource_type=False,
+                    require_resource_id=False,
+                )
 
-        try:
-            resource_ids = self._verify(
-                resource_type=resource_type,
-                resource_id=resource_id,
-            )
-        except AuthorizationException as exc:
-            resources.error = (
-                f"connector '{self.name}' authorization failure: {exc}"
-            )
-            # Log an exception if debug logging is enabled
-            if logger.isEnabledFor(logging.DEBUG):
-                logger.exception(resources.error)
-            else:
-                logger.warning(resources.error)
-            return resources
-        except Exception as exc:
-            error = (
-                f"connector {self.name} verification failed with unexpected "
-                f"error: {exc}"
-            )
-            # Log an exception if debug logging is enabled
-            if logger.isEnabledFor(logging.DEBUG):
-                logger.exception(error)
-            else:
-                logger.warning(error)
-            resources.error = (
-                "an unexpected error occurred while verifying the connector."
-            )
-            return resources
+                resource_ids = self._verify(
+                    resource_type=resource_type,
+                    resource_id=resource_id,
+                )
+            except ValueError as exc:
+                logger.debug(f"connector verification failed: {exc}")
+                raise ValueError(
+                    f"The connector configuration is incomplete or invalid: "
+                    f"{exc}",
+                )
+            except AuthorizationException as exc:
+                error = f"connector{name_msg} authorization failure: {exc}"
+                # Log an exception if debug logging is enabled
+                if logger.isEnabledFor(logging.DEBUG):
+                    logger.exception(error)
+                else:
+                    logger.warning(error)
 
-        if not resource_type:
-            # For multi-type connectors, if a resource type is not provided
-            # as argument, we don't expect any resources to be listed
-            return resources
+                # Log the error in the resources object
+                resources.set_error(error, resource_type=resource_type)
+                if resource_type:
+                    continue
+                else:
+                    # We stop on a global failure
+                    break
+            except Exception as exc:
+                error = (
+                    f"connector{name_msg} verification failed with "
+                    f"unexpected error: {exc}"
+                )
+                # Log an exception if debug logging is enabled
+                if logger.isEnabledFor(logging.DEBUG):
+                    logger.exception(error)
+                else:
+                    logger.warning(error)
```

Comment:
Seems odd, why?

Reply:
This is a choice I had to make to keep the CLI output cleaner. Basically, if you have a lot of connectors that spit out errors, you end up with screens full of exception traces and error messages before you get the final result, which can be really annoying. With this in place, you only get the final messages as warnings.